### PR TITLE
[Implement][Ccurtis]Data Collection Task. 

### PR DESCRIPTION
**Pull request by constantin73**
Created 5 classes. Created 2 objects of Order type.
Created a method that return only Orders containing item.type = "Clothes"
Created a method that return list of users who's City is Chisinau

### CODE REVIEW

**Review by IIachimov — DISMISSED** [review]

File: constantinCurtis/src/main/java/HomeWork/Classes/Address.java [hunk 1]
```diff
@@ -0,0 +1,26 @@
+package HomeWork.Classes;
```
Comment (solonarieugeniu) [inline constantinCurtis/src/main/java/HomeWork/Classes/Address.java]: please see naming conventions, all-lowercase

File: constantinCurtis/src/main/java/HomeWork/Classes/Address.java [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+public class Address {
+    public String country;
+    public String city;
+    public String street;
+
+    public String getFullAddress() {
+        return country + " , " + city + ", " + street;
```
Comment (solonarieugeniu) [inline constantinCurtis/src/main/java/HomeWork/Classes/Address.java]: try to use join()

File: constantinCurtis/src/main/java/data/collection/task/Main.java [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class Main {
+    public static void main(String[] args) {
+
+        //create items
+        List<Item> ItemFirstList = new ArrayList<>();
+        Item item1 = new Item("Jeans", Type.CLOTHES);
```
Comment (solonarieugeniu) [inline constantinCurtis/src/main/java/data/collection/task/Main.java]: try to be more suggestive

File: constantinCurtis/src/main/java/data/collection/task/Main.java [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+
+        //Create list of orders
+        List<Order> orderList = new ArrayList<>();
+        orderList.add(order1);
+        orderList.add(order2);
+
+        //Filter Orders by Type CLOTHES, calling function filterOrderByItemType(orderList)
+        List<Order> filteredOrderList = Order.filterOrderByItemType(orderList, Type.CLOTHES);
+        System.out.println("Number of orders:" + filteredOrderList.size());
+
```
Comment (solonarieugeniu) [inline constantinCurtis/src/main/java/data/collection/task/Main.java]: remove empty line

File: constantinCurtis/src/main/java/data/collection/task/Main.java [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+        // Creating the list for users
+        List<User> userList = new ArrayList<>();
+        userList.add(userFirst);
+        userList.add(userSecond);
+
+        // Filter users by city
+        List<User> filteredListofUsers = new ArrayList<>();
+        filteredListofUsers = User.filterUsersByCity(userList, "Chisinau"); // Looks for the users from Chisinau city
+        System.out.println("Number of users:" + filteredListofUsers.size());
+
```
Comment (solonarieugeniu) [inline constantinCurtis/src/main/java/data/collection/task/Main.java]: same

File: constantinCurtis/src/main/java/data/collection/task/Main.java [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+
+        // Creating the list for users
+        List<User> userList = new ArrayList<>();
+        userList.add(userFirst);
+        userList.add(userSecond);
+
+        // Filter users by city
+        List<User> filteredListofUsers = new ArrayList<>();
+        filteredListofUsers = User.filterUsersByCity(userList, "Chisinau"); // Looks for the users from Chisinau city
```
Comment (solonarieugeniu) [inline constantinCurtis/src/main/java/data/collection/task/Main.java]: static import

File: constantinCurtis/src/main/java/data/collection/task/Main.java [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+    public static void main(String[] args) {
+
+        //create items
+        List<Item> ItemFirstList = new ArrayList<>();
+        Item item1 = new Item("Jeans", Type.CLOTHES);
+        Item item2 = new Item("Ball", Type.TOY);
+        //create second Item List
+
+        List<Item> ItemSecondList = new ArrayList<>();
+        Item item3 = new Item("Samsung", Type.GADJET);
```
Comment (solonarieugeniu) [inline constantinCurtis/src/main/java/data/collection/task/Main.java]: static import

File: constantinCurtis/src/main/java/data/collection/task/Main.java [hunk 8]
```diff
@@ -0,0 +1,70 @@
...
+        Order order2 = new Order(userSecond, ItemSecondList);
+
+        //Create list of orders
+        List<Order> orderList = new ArrayList<>();
+        orderList.add(order1);
+        orderList.add(order2);
+
+        //Filter Orders by Type CLOTHES, calling function filterOrderByItemType(orderList)
+        List<Order> filteredOrderList = Order.filterOrderByItemType(orderList, Type.CLOTHES);
+        System.out.println("Number of orders:" + filteredOrderList.size());
```
Comment (solonarieugeniu) [inline constantinCurtis/src/main/java/data/collection/task/Main.java]: try to replace with Logger

File: constantinCurtis/src/main/java/README.md [hunk 9]
```diff
@@ -1,10 +0,0 @@
-1. The current repository contains 5 packages. 1 Package per student.
```
Comment (solonarieugeniu) [inline constantinCurtis/src/main/java/README.md:1]: deleted file, are you sure?

**Comment by solonarieugeniu** [comment]
do squash commits

File: constantinCurtis/src/main/java/HomeWork/Classes/Address.java (hunk 1 above)
Comment (constantin73) [reply constantinCurtis/src/main/java/HomeWork/Classes/Address.java]: Changed

File: constantinCurtis/src/main/java/HomeWork/Classes/Address.java (hunk 2 above)
Comment (constantin73) [reply constantinCurtis/src/main/java/HomeWork/Classes/Address.java]: join was used in Address Class

File: constantinCurtis/src/main/java/data/collection/task/Main.java (hunk 3 above)
Comment (constantin73) [reply constantinCurtis/src/main/java/data/collection/task/Main.java]: Changed name of the "item1"

File: constantinCurtis/src/main/java/data/collection/task/Main.java (hunk 4 above)
Comment (constantin73) [reply constantinCurtis/src/main/java/data/collection/task/Main.java]: done

File: constantinCurtis/src/main/java/data/collection/task/Main.java (hunk 5 above)
Comment (constantin73) [reply constantinCurtis/src/main/java/data/collection/task/Main.java]: done

File: constantinCurtis/src/main/java/README.md (hunk 9 above)
Comment (constantin73) [reply constantinCurtis/src/main/java/README.md:1]: Yes. File README.md should be deleted. Andrei Secu confirmed.

File: constantinCurtis/src/main/java/data/collection/task/Main.java (hunk 8 above)
Comment (constantin73) [reply constantinCurtis/src/main/java/data/collection/task/Main.java]: it was changed

File: constantinCurtis/src/main/java/data/collection/task/Main.java (hunk 7 above)
Comment (constantin73) [reply constantinCurtis/src/main/java/data/collection/task/Main.java]: I used here static import

File: constantinCurtis/src/main/java/data/collection/task/Main.java (hunk 6 above)
Comment (constantin73) [reply constantinCurtis/src/main/java/data/collection/task/Main.java]: I used here static import

**Review by IIachimov — DISMISSED** [review]

**Review by iMarari — DISMISSED** [review]

**Review by gaenache — DISMISSED** [review]

File: constantinCurtis/src/main/java/data/collection/task/classes/Order.java [hunk 10]
```diff
@@ -0,0 +1,52 @@
...
+import java.util.List;
+
+public class Order {
+    private static long idCounter = 0;
+    public long id;
+    public LocalDate date;
+    public User user;
+    public List<Item> items;
+
+    //cntrl+alt+l =arrangare de cod
```
Comment (asecu) [inline constantinCurtis/src/main/java/data/collection/task/classes/Order.java]: nice try...This is a part of your application? Please, store it somewhere else and let the Order.java contain only class related code.

File: constantinCurtis/src/main/java/data/collection/task/classes/Order.java [hunk 11]
```diff
@@ -0,0 +1,52 @@
...
+    public String toString() {
+        return "Order{" +
+                "id=" + id +
+                ", date=" + date +
+                ", user=" + user +
+                ", items=" + items +
+                '}';
+    }
+
+    // creating function to filter the orders by type CLOTHES
```
Comment (asecu) [inline constantinCurtis/src/main/java/data/collection/task/classes/Order.java]: In java we are operating with methods, NO Functions any more, therefore you should replace the function word with method.

File: constantinCurtis/src/main/java/data/collection/task/classes/User.java [hunk 12]
```diff
@@ -0,0 +1,47 @@
...
+    @Override
+    public String toString() {
+        return "User{" +
+                "id=" + id +
+                ", username='" + username + '\'' +
+                ", addresses=" + addresses +
+                '}';
+    }
+
+    // create function to filter list of users whose city is Chisinau
```
Comment (asecu) [inline constantinCurtis/src/main/java/data/collection/task/classes/User.java]: The function word should be changed

**Review by solonarieugeniu — DISMISSED** [review]

File: constantinCurtis/src/main/java/data/collection/task/classes/Order.java (hunk 10 above)
Comment (constantin73) [reply constantinCurtis/src/main/java/data/collection/task/classes/Order.java]: deleted the row 14

File: constantinCurtis/src/main/java/data/collection/task/classes/Order.java (hunk 11 above)
Comment (constantin73) [reply constantinCurtis/src/main/java/data/collection/task/classes/Order.java]: thanks, done

File: constantinCurtis/src/main/java/data/collection/task/classes/User.java (hunk 12 above)
Comment (constantin73) [reply constantinCurtis/src/main/java/data/collection/task/classes/User.java]: changed

**Review by asecu — APPROVED** [review]

**Review by IIachimov — APPROVED** [review]

**Review by iMarari — APPROVED** [review]

**Review by solonarieugeniu — APPROVED** [review]

**Pull request merged**